### PR TITLE
fix: use different error messages for namespace scoped token errors

### PR DESCRIPTION
**Pull request by rajyan**
Had some difficulty debugging the namespace scoped token error, so improved the error message reason.

### CODE REVIEW

File: internal/server/authn/middleware/grpc/middleware.go [hunk 1]
```diff
@@ -420,14 +420,14 @@ func NamespaceMatchingInterceptor(logger *zap.Logger, o ...containers.Option[Int
...
-						zap.String("reason", "namespace is not allowed"))
+						zap.String("reason", "same namespace is not set for all requests"))
 					return ctx, errUnauthenticated
 				}
 			}
 		default:
 			// if the the token has a namespace but the request does not then we should reject the request
 			logger.Error("unauthenticated",
-				zap.String("reason", "namespace is not allowed"))
+				zap.String("reason", "namespace is requried when using namespace scoped token"))
```
Comment (GeorgeMac) [inline internal/server/authn/middleware/grpc/middleware.go:430]: Small typo:
```suggestion
				zap.String("reason", "namespace is required when using namespace scoped token"))
```

Comment (rajyan) [reply internal/server/authn/middleware/grpc/middleware.go]: Thank you!
Fixed and rebased with signed-off commits.

**Review by GeorgeMac — APPROVED** [review]
Thank for this @rajyan 🙏 Nice improvement.

**Comment by markphelps** [comment]
@all-contributors please add @rajyan for code

**Pull request merged**